extensions/cl_ext_buffer_device_address.asciidoc

```diff
@@ -0,0 +1,320 @@
+  | This flag specifies that the buffer must have a single fixed address
+   for its lifetime and the address should be unique at least across the devices
+   of the context, but not necessarily within the host (virtual) memory.
+
+   The flag might imply that the buffer will be "pinned" permanently to
+   a device's memory, but might not be necessarily so, as long as the address
+   range of the buffer remains constant.
+
+   The address is guaranteed to remain the same until the buffer is freed, and
+   the address can be queried via {clGetMemObjectInfo}.
+
+   The device-specific buffer content updates are still performed by
+   implicit or explicit buffer migrations performed by the runtime or the
+   client code. If all of the devices in the context do not support
+   this type of allocations, an error (CL_INVALID_VALUE) is returned.
+
+   The device addresses of sub-buffers derived from CL_MEM_DEVICE_SHARED_ADDRESS_EXT
+   allocated buffers can be computed by adding the sub-buffer origin to the
+   start address.
+
+| {CL_MEM_DEVICE_PRIVATE_ADDRESS_EXT_anchor}
+
+include::{generated}/api/version-notes/CL_MEM_DEVICE_PRIVATE_ADDRESS_EXT.asciidoc[]
+  | This flag specifies that the buffer must have a single fixed address
+   for its lifetime. Each device in the context can have their own (fixed)
+   device-side address and a copy of the created buffer which are synchronized
+   implicitly by the runtime. The main difference to a default cl_mem allocation
+   in that case is that the addresses are queriable with CL_MEM_DEVICE_PTRS_EXT
+   and the per-device address is guaranteed to be the same for the entire lifetime
+   of the cl_mem.
+
+   The device addresses of sub-buffers derived from CL_MEM_DEVICE_PRIVATE_ADDRESS_EXT
+   allocated buffers can be computed by adding the sub-buffer origin to the
+   device-specific start address.
+
+|====
+
+// refError
+
+=== Section 5.5.6 - Memory Object Queries
+
+Add a new information type <<clGetMemObjectInfo, List of supported param_names table>>:
+
+[width="100%",cols="<33%,<17%,<50%",options="header"]
+|====
+| Memory Object Info | Return type | Description
+| {CL_MEM_DEVICE_PTR_EXT_anchor}
+
+include::{generated}/api/version-notes/CL_MEM_DEVICE_PTR_EXT.asciidoc[]
+  | {cl_mem_device_address_EXT_TYPE}
+      | Returns the device address for a buffer allocated with
+   CL_MEM_DEVICE_SHARED_ADDRESS_EXT. If the buffer was not created with the flag
+   or there are multiple devices in the context and the buffer address is
+   not the same for all of them, it returns CL_INVALID_MEM_OBJECT.
+
+| {CL_MEM_DEVICE_PTRS_EXT_anchor}
+include::{generated}/api/version-notes/CL_MEM_DEVICE_PTRS_EXT.asciidoc[]
+  | {cl_mem_device_address_pair_EXT_TYPE}
+      | Returns the device-address pairs for all devices in the context.
+      The per-device addresses might differ when the buffer was allocated
+      with the CL_MEM_DEVICE_PRIVATE_ADDRESS_EXT enabled.
```

Comment:
It might be simpler to just return an array of device addresses, with the ordering matching that of the devices in the context. You could also avoid using two separate queries. The array could contain multiple times the same value.

Reply:
Good idea. Changed.